### DISCUSSION TITLE
Drop `do` keyword?

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#8704] [Ideas]
We currently have to call an effect handler op with `do`.

But this gets silly when we have convenience methods.

Take for example:

```scala
pub eff Logger {
    pub def log(s: Severity, m: String): Unit
}

mod Logger {
    pub def trace(m: a): Unit \ Logger with ToString[a] = do Logger.log(Severity.Trace, "${m}")
}
```

It seems the most obvious solution is to drop it. The alternative would be to require it for all *control affecting expressions*. But then again, why do that?

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup DO we agree to drop it? (see what I did there?)

**Reply by mlutze** [reply]
I agree with the proposal, but I think a censure is in order for the joke.

**Reply by JonathanStarup** [reply]
sounds good. there still have to be an internal concept though for the backend

**Reply by mlutze** [reply]
Resolver will handle that

**Comment by jaschdoc** [comment]
Sound like a good thing to `do`.

**Comment by magnus-madsen** [comment]
We are dropping it.
